### PR TITLE
Query multiple tables

### PR DESCRIPTION
**Pull request by abenaTenkorang**
Changes introduced ♻️
Create a table named owners ✔
Create a table named species ✔
Modify animals table ✔
Insert data to the owners and species table ✔
Use the primary key and foreign key mechanism for joining tables ✔
Query multiple tables ✔
Prepare complex queries that answer analytical questions ✔
Screenshots of the analytical queries.
1. What animals belong toMelody Pond?

Screenshot 2023-05-04 230736

2. List all animals that are Pokeman.
Screenshot 2023-05-04 211125

3. List all owners and their animals.
Screenshot 2023-05-04 211213

4. List all Digimon owned by Jennifer Orwell.

Screenshot 2023-05-04 211450

The query returned 0 rows.
5. Who owns the most animals?

Screenshot 2023-05-04 211552

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -6,5 +6,24 @@ CREATE TABLE animals (
     date_of_birth date not null,
     escape_attempts int not null,
     neutered boolean not null,
-    weight_kg decimal not null
+    weight_kg decimal not null,
+    species varchar(100)
 );
```
Comment (Kamba56) [inline schema.sql:11]: - [x] As demanded by project requirements, kindly remove the species column from the animals' table. You can use `ALTER TABLE  REMOVE species;` command.

File: data.sql [hunk 2]
```diff
@@ -1,6 +1,43 @@
...
+                    WHEN name LIKE '%mon' THEN 2 
+                    ELSE 1 
+                 END,
+    owner_id = CASE 
+                    WHEN name = 'Agumon' THEN 1 
+                    WHEN name IN ('Gabumon', 'Pikachu') THEN 2 
+                    WHEN name IN ('Devimon', 'Plantmon') THEN 3 
+                    WHEN name IN ('Charmander', 'Squirtle', 'Blossom') THEN 4 
+                    WHEN name IN ('Angemon', 'Boarmon') THEN 5 
+                 END;
```
Comment (Kamba56) [inline data.sql]: - [x] Kindly add a command that updates the species_id column of your animals' table. If an animal's name ends with mon it should have Digimon as a species_id, all other animals should have Pokemon instead. Add the command to your data.sql file.

File: queries.sql [hunk 3]
```diff
@@ -14,3 +14,98 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg < 10.5;
...
+
+
+-- What animals belong to Melody Pond?
+SELECT name 
+FROM animals 
+WHERE owner_id = (
+    SELECT id 
+    FROM owners 
+    WHERE full_name = 'Melody Pond'
+);
```
Comment (Kamba56) [inline queries.sql]: - [ ] Melody Pond has three animals, Chamander, squirtle and blossom but your query from your screenshots only returns two, kindly fix this please.

File: queries.sql [hunk 4]
```diff
@@ -14,3 +14,98 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg < 10.5;
...
+SELECT full_name, name 
+FROM owners 
+LEFT JOIN animals ON owners.id = animals.owner_id 
+ORDER BY full_name;
+
+-- How many animals are there per species?
+SELECT species.name, COUNT(*) 
+FROM animals 
+JOIN species ON animals.species_id = species.id 
+GROUP BY species.name;
```
Comment (Kamba56) [inline queries.sql]: - [x] Kindly add the screenshot of the result from this query.

File: queries.sql [hunk 5]
```diff
@@ -14,3 +14,98 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg < 10.5;
...
+FROM animals 
+JOIN owners ON animals.owner_id = owners.id 
+JOIN species ON animals.species_id = species.id 
+WHERE owners.full_name = 'Jennifer Orwell' AND species.name = 'Digimon';
+
+-- List all animals owned by Dean Winchester that haven't tried to escape.
+SELECT name 
+FROM animals 
+JOIN owners ON animals.owner_id = owners.id 
+WHERE owners.full_name = 'Dean Winchester' AND escaped_attempts = false;
```
Comment (Kamba56) [inline queries.sql]: - [x] Kindly add a screenshot of the response from this query.

File: queries.sql [hunk 6]
```diff
@@ -14,3 +14,98 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg < 10.5;
...
+
+-- Who owns the most animals?
+SELECT owners.full_name, COUNT(animals.name) 
+FROM owners 
+LEFT JOIN animals ON owners.id = animals.owner_id 
+GROUP BY owners.full_name 
+ORDER BY COUNT(animals.name) DESC 
+LIMIT 1;
```
Comment (Kamba56) [inline queries.sql]: - [x] Melody pond has the most animals which are three in number. Kindly fix this query, please.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @abenaTenkorang, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great job adding owners and species table and adding a one-to-many relationship to animals table.!👏👏👏
- [x] 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 6 above)
Comment (abenaTenkorang) [reply queries.sql]: Screenshot 2023-05-04 230736

File: queries.sql (hunk 5 above)
Comment (abenaTenkorang) [reply queries.sql]: Screenshot 2023-05-04 230628

File: data.sql (hunk 2 above)
Comment (abenaTenkorang) [reply data.sql]: Screenshot 2023-05-04 230236


Screenshot 2023-05-04 230323

File: schema.sql (hunk 1 above)
Comment (abenaTenkorang) [reply schema.sql:11]: Screenshot 2023-05-04 231418

File: queries.sql (hunk 3 above)
Comment (abenaTenkorang) [reply queries.sql]: Screenshot 2023-05-04 231548

File: data.sql (hunk 2 above)
Comment (abenaTenkorang) [reply data.sql]: Thank you very much for pointing out all I omitted @Kamba56

File: queries.sql (hunk 4 above)
Comment (abenaTenkorang) [reply queries.sql]: Screenshot 2023-05-04 230515

File: schema.sql [hunk 7]
```diff
@@ -6,5 +6,24 @@ CREATE TABLE animals (
...
+
+-- Create the species table
+CREATE TABLE species (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(255)
+);
+
+-- Modify the animals table
+ALTER TABLE animals
+ADD COLUMN species_id INTEGER REFERENCES species(id),
```
Comment (rloterh) [inline schema.sql]: - [ ] We want to expand the _species_ column found in _animals_ table by creating a _species_ table. Therefore we need to remove the _species_ column of from _animals_ table and rather link _species_ table to _animal_ table using a foreign key. 

     Good job for creating the _species_ table and adding foreign key reference, therefore let's remove the _species_ column from the _animals_ table by adding this query just after _line 27_:

```sql

REMOVE species;

```

File: queries.sql (hunk 3 above)
Comment (rloterh) [reply queries.sql]: Needs attention!
Screenshots should be updated within the PR description.

File: queries.sql (hunk 4 above)
Comment (rloterh) [reply queries.sql]: Needs attention!
Screenshots should be updated within the PR description.

File: queries.sql (hunk 5 above)
Comment (rloterh) [reply queries.sql]: Needs attention!
Screenshots should be updated within the PR description.

File: queries.sql (hunk 6 above)
Comment (rloterh) [reply queries.sql]: Needs attention!
Screenshots should be updated within the PR description.

File: data.sql (hunk 2 above)
Comment (rloterh) [reply data.sql]: Screenshots should be updated within the PR description.

File: schema.sql (hunk 1 above)
Comment (rloterh) [reply schema.sql:11]: Screenshots should be updated within the PR description.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Abi, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 

## Highlights 👍  
- Good job so far, queries are coming up pretty well! 👏 

## Required Changes ♻️
_Kindly check the inline comments above/below the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

**You can also consider:**
- Kindly ensure that provided placeholder texts/ guide instructions are removed and/or replaced with appropriate contents applicable under each section of your ReadMe. Similarly, kindly also remove guide notes for _license_ creation under the _license_ section.

**Kindly remove all similar to the underlined:**
image

Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 8]
```diff
@@ -14,3 +14,82 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg < 10.5;
...
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
+
+
+-- what animal belong to melody pond?
+select full_name, name from owners 
+join animals on animals.owners_id = owners.id where owners.full_name = 'Melody Pond';
+
```
Comment (danmainah) [inline queries.sql:64]: - [ ] @abenaTenkorang great work implementing the changes requested, however as the previous reviewer noted you have not updated the query screenshot in the pull request description, kindly consider updating it
image

**Review by danmainah — CHANGES_REQUESTED** [review]

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @abenaTenkorang , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ Professional README file.
✔️ Github flow was correctly used

 # Required Changes ♻️
- [x] @abenaTenkorang great work implementing the changes requested, however as the previous reviewer noted you have not updated the query screenshot in the pull request description, kindly consider updating it
image

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by abenaTenkorang** [comment]
Please I have made all changes requested by all 3 reviewers and this is my extra review. Thank you

**Review by Munsa1 — APPROVED** [review]
Hi @abenaTenkorang, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Updated Screenshots in PR as Requested :+1: 

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**